d3a_api_client/redis_device.py

```diff
@@ -11,11 +11,15 @@ def __init__(self, device_id, autoregister=True, redis_url='redis://localhost:63
 
     def _on_register(self, msg):
         message = json.loads(msg["data"])
+        if not self._is_trans_id_matching(message):
+            return
```

Comment:
I think it is better if you raise an exception here. That way the user will be notified that the registration failed, and the client will exit.

Reply:
You are right! There is only ONE register request in this channel. Will add it.

Reply:
But this applies also to the rest client, right? Will add it there too.

Reply:
Applies to the rest client indeed. Thanks